module/data/item/templates/activities.mjs

```diff
@@ -333,6 +334,109 @@ export default class ActivitiesTemplate extends SystemDataModel {
     return { updates, rolls };
   }
 
+  /* -------------------------------------------- */
+  /*  Socket Event Handlers                       */
+  /* -------------------------------------------- */
+
+  /**
+   * Perform any necessary actions when an item with activities is created.
+   * @param {object} data     The initial data object provided to the document creation request.
+   * @param {object} options  Additional options which modify the update request.
+   * @param {string} userId   The id of the User requesting the document update.
+   */
+  async onCreateActivities(data, options, userId) {
+    if ( (userId !== game.user.id) || !this.parent.isEmbedded ) return;
+
+    // If item has any Cast activities, create locally cached copies of the spells
+    const spells = (await Promise.all(this.activities.getByType("cast").map(a => a.getCachedSpellData())))
+      .filter(_ => _);
+    if ( spells.length ) this.parent.actor.createEmbeddedDocuments("Item", spells);
+  }
+
+  /* -------------------------------------------- */
+
+  /**
+   * Prepare any item or actor changes based on activity changes.
+   * @param {object} changed  The differential data that is changed relative to the documents prior values.
```

Comment:
```suggestion
   * @param {object} changed  The differential data that is changed relative to the document's prior values.
```

Reply:
That's what I get for grabbing JSDocs straight from core :P

Reply:
(: